lib/pbench/agent/tool_data_sink.py

```diff
@@ -762,10 +763,16 @@ def fetch_params(params, pbench_run):
                 tool_metadata=ToolMetadata.tool_md_from_dict(params["tool_metadata"]),
                 tool_trigger=params["tool_trigger"],
                 tools=params["tools"],
+                instance_uuid=params["instance_uuid"],
             )
         except KeyError as exc:
             raise ToolDataSinkError(f"Invalid parameter block, missing key {exc}")
         else:
+            if instance_uuid != tdsp.instance_uuid:
+                raise ToolDataSinkError(
+                    f"Parameter block has unexpected UUID '{tdsp.instance_uuid}',"
+                    f" expected '{instance_uuid}'"
+                )
```

Comment:
Why is this code here instead of in the caller?  In order to have it here, we have to add a parameter to the function signature.  It would be simpler (and less coupled) to just check in the caller.

Reply:
But we there are two callers and we wanted the knowledge of valid parameters in one place.